lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,326 @@
+<?php
+/**
+ * REST API: WP_REST_Block_Types_Controller class
+ *
+ * @since      5.4.0
+ * @subpackage REST_API
+ * @package    WordPress
+ */
+
+/**
+ * Core class used to access block types via the REST API.
+ *
+ * @see   WP_REST_Controller
+ */
+class WP_REST_Block_Types_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'block-types';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see   register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<namespace>[a-zA-Z0-9_-]+)/(?P<name>[a-zA-Z0-9_-]+)',
+			array(
+				'args'   => array(
+					'name'      => array(
+						'description' => __( 'Block name', 'gutenberg' ),
+						'type'        => 'string',
+					),
+					'namespace' => array(
+						'description' => __( 'Block namsspace', 'gutenberg' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read post block types.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
For what reason is the PHPCS rule ignored here?

Reply:
`$request` is not used but is required to be passed.

Reply:
> `$request` is not used but is required to be passed.

Can you clarify what you mean by "required"? Will something break if...

```suggestion
	public function get_items_permissions_check() {
```

Reply:
Ah, I see now:

>Warning: Declaration of WP_REST_Block_Types_Controller::get_items_permissions_check() should be compatible with WP_REST_Controller::get_items_permissions_check($request) in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-rest-block-types-controller.php on line 0

If my experience demonstrates anything, it could be an easy thing to overlook or be confused about in future maintenance. At least in Gutenberg's JavaScript code, there's a tendency to rationalize any rule disabling for this reason (also idea for formalized enforcement at #16795).

Not sure it's quite as easy to do here in this context, but worth considering.

Reply:
VariableAnalysis lint is not part of the core lints. That ignore is just for this plugin. I will be removed once in core.